python/pyarrow/array.pxi

```diff
@@ -1778,6 +1778,49 @@ cdef class Array(_PandasConvertible):
 
         return pyarrow_wrap_array(array)
 
+    def __dlpack__(self, stream=None):
+        """Export a primitive array as a DLPack capsule.
+
+        Parameters
+        ----------
+        stream : int, optional
+            A Python integer representing a pointer to a stream. Currently not supported.
+            Stream is provided by the consumer to the producer to instruct the producer
+            to ensure that operations can safely be performed on the array.
+
+        Returns
+        -------
+        capsule : PyCapsule
+            A DLPack capsule for the array, pointing to a DLManagedTensor.
+        """
+        cdef CResult[DLManagedTensor*] c_dlm_tensor
+
+        if stream is None:
+            c_dlm_tensor = ExportToDLPack(pyarrow_unwrap_array(self))
+            dlm_tensor = GetResultValue(c_dlm_tensor)
+
+            return PyCapsule_New(dlm_tensor, 'dltensor', dlpack_pycapsule_deleter)
+        else:
+            raise NotImplementedError(
+                "Only stream=None is supported."
+            )
+
+    def __dlpack_device__(self):
+        """
+        Performs the operation __dlpack_device__.
+
+        Returns
+        -------
+        tuple : Tuple[DLDeviceType, int]
```

Comment:
Does it really return a "DLDeviceType"? In the tests, the result is compared to an int.

Reply:
Yeah, it currently returns an int.

Numpy returns a `tuple[int, Literall[0]]`: https://github.com/numpy/numpy/blob/0be4154648cd6e90aae32246349085f7ca8f2b13/numpy/typing/tests/data/reveal/ndarray_misc.pyi#L34

but the documentation is clear about returning `Tuple[Enum, int]`:
https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__dlpack_device__.html

Will update code to follow the array-api docs.

Reply:
Decided to do the same as Numpy at the end - returning `tuple[int, int]` =)
No need to complicate if it is not necessary.